code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -895,6 +895,10 @@ func (d *BlobberAllocation) delete(size int64, now common.Timestamp,
 	rdtu float64) (move currency.Coin) {
 
 	move = currency.Coin(sizeInGB(size) * float64(d.Terms.WritePrice) * rdtu)
+	if move > d.ChallengePoolIntegralValue {
```

Comment:
most likely it can be used as a hack, but can't be a solution, why move is bigger than `ChallengePoolIntegralValue`, it should be calculated the way it is not bigger anytime

Reply:
This is not a hack. It is possible that challenge response txn time is slightly before delete WM txn time. In that case we will have get negative tokens and same issue can happen. This is just for safety. Actual fix is here : https://github.com/0chain/0chain/pull/2867/files#diff-29bc2103d9eaa2e36a299ee645fdb44e5dd0cc52fc614a861d448108716163d8R807-R811

Reply:
> This is not a hack. It is possible that challenge response txn time is slightly before delete WM txn time.

And why is it the problem? challenge response has a round when it was generated, delete WM has some time after it, how can `ChallengePoolIntegralValue` become smaller than move? With suggested assignment we will make total writepool to be smaller than we need for final calculations